src/jobs/models/manager.ts

```diff
@@ -138,34 +161,129 @@ export class JobManager {
     await this.prisma.job.update(updateQueryBody);
   }
 
+  public async deleteJob(jobId: string): Promise<void> {
+    const deleteStages = this.prisma.stage.deleteMany({
+      where: {
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        job_id: jobId,
+      },
+    });
+
+    const deleteJob = this.prisma.job.delete({
+      where: {
+        id: jobId,
+      },
+    });
+
+    try {
+      await this.prisma.$transaction([deleteStages, deleteJob]);
+    } catch (err) {
+      if (err instanceof Prisma.PrismaClientKnownRequestError && err.code === prismaKnownErrors.recordNotFound) {
+        throw new JobNotFoundError(JOB_NOT_FOUND_MSG);
+      }
+
+      throw err;
+    }
+  }
+
+  public async addStages(jobId: string, stagesPayload: StageCreateModel[]): Promise<JobModel> {
```

Comment:
1. I would choose a different approach of inserting the stages straight into prisma.stage instead of modifying the job.
2. As you are adding a stage, it should be part of stageManager.

Reply:
all done :)